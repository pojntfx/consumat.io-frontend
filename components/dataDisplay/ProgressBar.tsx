import { useEffect, useState } from "react";
import CSS from "csstype";

type ProgressBarProps = {
  progress: number;
  limit: number;
  className?: string;
};

function ProgressBar({ progress, limit, className }: ProgressBarProps) {
  const [progressBar, setProgressBar] = useState<CSS.Properties>();
  useEffect(() => {
    setProgressBar({ width: `${(100 / limit) * progress}%` });
  }, [progress, limit]);
  return (
    <div
      className={
        "w-full h-3 rounded-full overflow-hidden bg-gray-200 " + className
      }
    >
      <div
        className="h-full bg-gradient-to-br from-green-500 to-blue-500"
        style={progressBar}
      ></div>
    </div>
  );
}

export default ProgressBar;
