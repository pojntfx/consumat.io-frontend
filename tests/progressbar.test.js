import React from "react";
import renderer from "react-test-renderer";
import ProgressBar from "../components/dataDisplay/ProgressBar";
import testExamples from "./testExamples";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ProgressBar
        progress={testExamples.exampleProgress}
        limit={testExamples.exampleLimit}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
