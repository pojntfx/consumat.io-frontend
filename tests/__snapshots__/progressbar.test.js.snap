// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-full h-3 rounded-full overflow-hidden bg-gray-200 undefined"
>
  <div
    className="h-full bg-gradient-to-br from-green-500 to-blue-500"
  />
</div>
`;
